Proiect1/Proiect1/Animals/AnimalBase.cs

```diff
@@ -0,0 +1,11 @@
+﻿using Proiect1.Interface;
+
+namespace Proiect1.Animals
+{
+    public abstract class AnimalBase : IAnimal
```

Comment:
do you need this class? I like the usage of an interface, but is this class needed?

Reply:
the short answer: no, this application can be made without an AnimalBase class
the long answer: to be honest, the initial solution I sent is an application I made two years ago, I sent it in its most natural form because in the requirements was mentioned the fact that I need to do at least a refactoring. Now related to the AnimalBase class the answer would be "yes" to some extent. In the initial request, from what I remember, this was requested because base classes are used in cases where there are several functionalities or properties that can be used by other classes. At the same time, they are used to respect the DRY principle

after I looked more closely at the requirement, I think that this application can be solved in a much simpler way, like the oop examples on the w3schools website